pkg/api/v1beta1/types.go

```diff
@@ -397,6 +397,23 @@ type EnvVar struct {
 	Key  string `json:"key,omitempty" description:"name of the environment variable; must be a C_IDENTIFIER; deprecated - use name instead"`
 	// Optional: defaults to "".
 	Value string `json:"value,omitempty" description:"value of the environment variable; defaults to empty string"`
+	// Optional: specify a source for the value of the EnvVar.
+	Source *EnvVarSource `json:"source,omitempty" description:"source for the environment variable's value; cannot be used if value is not empty"`
+}
+
+// EnvVarSource represents a source for the value of an EnvVar.
+// Only one of its members may be specified.
+type EnvVarSource struct {
+	// Selects a field of the pod; only name and namespace are supported.
+	PodField *ObjectFieldSelector `json:"podField,omitempty" description:"selectes a field of the pod; only name and namespace are supported"`
+}
+
+// ObjectFieldSelector selects an APIVersioned field of an object.
+type ObjectFieldSelector struct {
+	// The API version the FieldPath is written in terms of.
+	APIVersion string `json:"apiVersion,omitempty" description="The API version that FieldPath is written in terms of"`
```

Comment:
regarding ObjectRef vs ObjectFieldSelector, here's my thinking.

It is a strict subset of ObejctRef.  We should not reuse struct just because it exists.  If it grows to be identical to ObjectRef, we can swap in that struct at will, and it will be compatible.  We would not add fields to a struct just because we might need them.  I feel we should be doing this MORE - use purpose-built structs that fit exactly what we need.  The struct name is not really even part of the API.

That's the last I'll say, if Brian wants to use Objectref, just do that